R/zchunk_LA154.Transport_china.R

```diff
@@ -0,0 +1,196 @@
+        # We will next reintroduce those rows using "complete" and assign those values to be 0
+        complete(nesting(GCAM_region_ID, UCD_sector, mode, size.class, UCD_technology, UCD_fuel, fuel), year = HISTORICAL_YEARS, fill = list(value = 0)) %>%
+        # Fuel and mode will be mapped to NBS fuel and sector
+        left_join_error_no_match(trnUCD_NBS_mapping, by = c("fuel", "mode")) ->
+        L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh
+
+
+      #next, extract the relevant NBS sector & fuel combinations from the full province database
+      L101.NBS_use_all_Mtce %>%
+        filter(year %in% HISTORICAL_YEARS) %>% # Ensure within historical period
+        mutate(fuel_sector = paste(EBProcess, EBMaterial)) %>% # Create concatenated list in base dataframe to match the syntax of our list above
+        filter(fuel_sector %in% paste(L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh$EBProcess, L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh$EBMaterial)) %>% # Filtering for just NBS-fuel/sector pairs
+        select(province, EBProcess, EBMaterial, sector, fuel, year, value) %>%
+        group_by(province, year) %>%
+        # First zero out NAs in years where some values are NA but not all
+        mutate(value = replace(value, is.na(value) & sum(value, na.rm = T) != 0, 0)) %>%
+        ungroup %>%
+        # use approx_fun rule = 2 to fill out data in years where the entire province is NA
+        group_by(province, EBProcess, EBMaterial) %>%
+        mutate(value = approx_fun(year, value, rule = 2)) %>%
+        ungroup() ->
+        L154.NBS_trn_Mtce_province
+
+      # To calculate the province share, we need to calculate the national amount
+      L154.NBS_trn_Mtce_province %>%
+        group_by(EBProcess, EBMaterial, year) %>% # Dropping province
+        summarise(value_national = sum(value, na.rm = T)) %>%
+        ungroup() ->
+        NBS_transportation_national
+
+      # Now the province shares can be calculated by dividing the province data by the national
+      L154.NBS_trn_Mtce_province %>%
+        left_join_error_no_match(NBS_transportation_national, by = c("EBProcess", "EBMaterial", "year")) %>%
+        mutate(value_share = value / value_national) %>% # Calculating province's share
+        # NAs were introduced where national values were 0. Replace NAs with zeros.
+        replace_na(list(value_share = 0)) %>%
+        select(province, EBProcess, EBMaterial, year, value_share) ->
+        L154.NBS_trn_share_province
+
+      # Match these percentages into a table of all transportation technologies that will be written out
+      L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh %>%
+        repeat_add_columns(tibble::tibble(province = gcamchina.PROVINCES)) ->
+      L154.pct_province_trn_m_sz_tech_F_Yh
+
+      # TODO: Tibet
+      #L154.pct_province_trn_m_sz_tech_F_Yh <- subset( L154.pct_province_trn_m_sz_tech_F_Yh, province != "XZ" )
+
+      L154.pct_province_trn_m_sz_tech_F_Yh %>%
```

Comment:
Move this section above the `TODO: Tibet` as one pipe.

Reply:
This is actually redundant. Why not just `full_join` the province share tibble and the national tibble? - national value will repeat for each province, and multiply national value with share.